rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -144,6 +197,101 @@ public ResponseEntity createReleases(
+
+        final List<VulnerabilityDTO> allVulnerabilityDTOs = this.vulnerabilityService.getVulnerabilitiesByProjectId(id, sw360User);
+
+        final List<Resource<VulnerabilityDTO>> vulnerabilityResources = new ArrayList<>();
+        for (final VulnerabilityDTO vulnerabilityDTO : allVulnerabilityDTOs) {
+            final Resource<VulnerabilityDTO> vulnerabilityDTOResource = new Resource<>(vulnerabilityDTO);
+            vulnerabilityResources.add(vulnerabilityDTOResource);
+        }
+        final Resources<Resource<VulnerabilityDTO>> resources = new Resources<>(vulnerabilityResources);
+        return new ResponseEntity<>(resources, HttpStatus.OK);
+    }
+
+    @RequestMapping(value = PROJECTS_URL + "/{id}/licenses", method = RequestMethod.GET)
+    public ResponseEntity<Resources<Resource<License>>> getLicensesOfReleases(@PathVariable("id") String id, OAuth2Authentication oAuth2Authentication) {
+        final User sw360User = this.restControllerHelper.getSw360UserFromAuthentication(oAuth2Authentication);
+        final Project project = this.projectService.getProjectForUserById(id, sw360User);
+        final List<Resource<License>> licenseResources = new ArrayList<>();
+        final Set<String> allLicenseIds = new HashSet<>();
+
+        final Map<String, ProjectReleaseRelationship> releaseIdToUsage = project.getReleaseIdToUsage();
+        for (final String releaseId : releaseIdToUsage.keySet()) {
+            final Release release = this.releaseService.getReleaseForUserById(releaseId, sw360User);
+            final Set<String> licenseIds = release.getMainLicenseIds();
+            allLicenseIds.addAll(licenseIds);
+        }
+        for (final String licenseId : allLicenseIds) {
+            final License license = this.licenseService.getLicenseById(licenseId);
+            license.setText(null);
+            license.setShortname(null);
+            final Resource<License> licenseResource = new Resource<>(license);
+            licenseResources.add(licenseResource);
+        }
+        final Resources<Resource<License>> resources = new Resources<>(licenseResources);
+
+        return new ResponseEntity<>(resources, HttpStatus.OK);
+    }
+
+    @RequestMapping(value = PROJECTS_URL + "/{id}/Releases/Ecc", method = RequestMethod.GET)
+    public ResponseEntity<Resources<Resource<Release>>> getECCsOfReleases(@PathVariable("id") String id, OAuth2Authentication oAuth2Authentication) {
+        final User sw360User = this.restControllerHelper.getSw360UserFromAuthentication(oAuth2Authentication);
+        final Project project = this.projectService.getProjectForUserById(id, sw360User);
+
+        final List<Resource<Release>> releaseResources = new ArrayList<>();
+
+        final Map<String, ProjectReleaseRelationship> releaseIdToUsage = project.getReleaseIdToUsage();
+        for (final String releaseId : releaseIdToUsage.keySet()) {
+            final Release release = this.releaseService.getReleaseForUserById(releaseId, sw360User);
```

Comment:
I would prefer to use variable names like releasesFromDatabase or sw360Releases to have a differential between the sw360 thrift objects and rest objects.